Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionSidebar.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <Foundation/Foundation.h>
+#import <WebKit/WKFoundation.h>
+
+@class WKWebExtensionContext;
+@class WKWebView;
+@protocol WKWebExtensionTab;
+@protocol WKWebExtension;
+
+WK_HEADER_AUDIT_BEGIN(nullability, sendability)
+
+/*!
+ @abstract This delegate notifies the browser that properties relevant to a currently-open sidebar have changed,
+ @discussion Implementing this delegate will allow the browser to react to changes to the sidebar object stemming from JavaScript API usage relating to the sidebar's web view or title.
+ */
+WK_CLASS_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA))
+NS_SWIFT_NAME(WKWebExtension.ActiveSidebarDelegate)
+@protocol _WKWebExtensionActiveSidebarDelegate <NSObject>
+
+/*!
+ @abstract Called when the sidebar's web view has been updated.
+ @discussion This method is called when the sidebar's existing ``WKWebView`` is replaced by a different ``WKWebView``, in which case the browser
+ must query the sidebar in order to obtain a reference to this new ``WKWebView`` to display. This method is not necessarily called when the sidebar's panel
+ URL is updated: if the URL can be updated without replacing the ``WKWebView``, it will be.
+ */
+- (void)webViewWasUpdated;
```

Comment:
What triggers a web view to change vs the URL just navigating the existing web view?

See comment about a view controller to help with this.